src/main/java/at/hannibal2/skyhanni/features/inventory/ItemPickupLog.kt

```diff
@@ -0,0 +1,314 @@
+            return true
+        }
+
+        if (bannedItemsConverted.contains(item.getInternalNameOrNull())) {
+            return true
+        }
+
+        if (item.getExtraAttributes()?.hasKey("quiver_arrow") == true) {
+            return true
+        }
+        return false
+    }
+
+    private fun updateItem(hash: Int, itemInfo: UpdatedItem, item: ItemStack, removed: Boolean) {
+        if (removed) {
+            itemsAddedToInventory[hash]?.let { added ->
+                added.time = SimpleTimeMark.now()
+            }
+            itemsRemovedFromInventory[hash]?.let {
+                it.updateAmount(itemInfo.amount)
+                return
+            }
+            if (isBannedItem(item)) return
+            itemsRemovedFromInventory[hash] = itemInfo
+        } else {
+            itemsRemovedFromInventory[hash]?.let { added ->
+                added.time = SimpleTimeMark.now()
+            }
+            itemsAddedToInventory[hash]?.let {
+                it.updateAmount(itemInfo.amount)
+                return
+            }
+            if (isBannedItem(item)) return
+            itemsAddedToInventory[hash] = itemInfo
+        }
+    }
+
+    private data class UpdatedItem(val name: String, var amount: Long, val neuInternalName: NEUInternalName?) {
+        var time = SimpleTimeMark.now()
+
+        fun updateAmount(change: Long) {
+            amount += change
+            time = SimpleTimeMark.now()
+        }
+
+        fun isExpired() = time.passedSince() > config.expireAfter.seconds
+    }
+
+    private fun isEnabled() = LorenzUtils.inSkyBlock && config.enabled
+
+    private fun renderList(prefix: String, amount: Long, name: String, itemIcon: NEUInternalName?): Renderable {
+
+
+        return Renderable.horizontalContainer(
+            buildList {
+                for (item in config.displayLayout) {
+                    when (item) {
+                        DisplayLayout.ICON -> {
+                            if (itemIcon != null) {
+                                addItemStack(itemIcon)
+                            } else {
+                                ItemNameResolver.getInternalNameOrNull(name)?.let { addItemStack(it) }
+                            }
+                        }
+
+                        DisplayLayout.CHANGE_AMOUNT -> {
+                            val formattedAmount = if (config.shorten) amount.shortFormat() else amount.addSeparators()
+                            add(Renderable.string("${prefix}${formattedAmount}"))
+                        }
+
+                        DisplayLayout.ITEM_NAME -> {
+                            add(Renderable.string(name))
+                        }
+
+                        null -> {}
+                    }
+                }
+            },
+        )
+    }
+
+
+    private fun worldChangeCooldown(): Boolean {
+        return LorenzUtils.lastWorldSwitch.passedSince() > 2.seconds
+    }
+
+    private fun updateDisplay() {
```

Comment:
extract logic into more functions, with names that explain their steps

Reply:
resolved thanks to j10a1n15